Classes/Domain/ExtractionManager.php

```diff
@@ -140,4 +140,31 @@ public function extractMetaData(Asset $asset) : MetaDataCollection
 
         return $metaDataCollection;
     }
+
+    /**
+     * @param Asset $asset
+     * @return MetaDataCollection
+     * @throws ExtractorException
+     * @throws UnknownObjectException
+     */
+    public function onAssetRemoved(Asset $asset) : MetaDataCollection
+    {
+        if ($asset instanceof ImageVariant) {
+            $asset = $asset->getOriginalAsset();
+        }
+
+        $flowResource = $asset->getResource();
+        if ($flowResource === null) {
+            throw new ExtractorException('Resource of Asset "' . $asset->getTitle() . '"" not found.', 1484060541);
+        }
+
+        $metaDataCollection = new MetaDataCollection();
+        $this->buildAssetMetaData($asset, $metaDataCollection);
+
+        // Because the signal assetRemoved was emitted the deleted flag for $flowResource is true
+        // thus the signal metaDataCollectionUpdated finally deletes the metadata for the asset
+        $this->metaDataManager->emitMetaDataCollectionUpdated($asset, $metaDataCollection);
+
+        return $metaDataCollection;
```

Comment:
Why return anything here? Slot return values are not used anyway…

Reply:
Didn't know that. Many thanks for the hint. I removed the return statement and corresponding comment.